### DISCUSSION TITLE
Sorting models fields

### DISCUSSION THREAD

**Discussion by wanjas** [prisma/prisma#16645] [Q&A] [answered]
Hello, sorry if I missed a solution for the issue somewhere. 

The use case is like that:
Our team uses `knex` to do migrations (in postgresql) and `db pull` to update the schema file. But for example, if two developers add two columns to the table in two different migrations they usually end up having these migrations applied in a different order to their development DBs. As a result, every time they commit new changes to the schema, the schema file is effectively the same but fields in the model are in a different order which leads to annoying conflict resolutions and unnecessary changes back and forth in the schema file. 

`prisma format` does a very good job of removing other discrepancies but isn't helping with this issue, unfortunately. 

Is there any way to sort fields in models in schema file alphabetically (or in any other consistent way)?

Any other advice on the issue is very much appreciated. 
Thank you

**Comment by nurul3101** [comment, marked as answer]
Hello @wanjas 👋 

At the moment, it's not possible to control the re-ordering behaviour. Here are a few related GitHub Issues:

- https://github.com/prisma/prisma/issues/2790
- https://github.com/prisma/prisma/issues/8283

I would recommend you to have a look at it and add a 👍 to them so that our product team can prioritise it.

Also, did you had a chance to look at this guide for [Team Development with Prisma Migrate](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/team-development)?

**Reply by wanjas** [reply]
Thank you for pointing out these GitHub issues and the guide. :+1: 

The decision to make migrations on DB first and introspect after that has been made when `prisma migrate` has been in early beta, we may reconsider it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
